compiler/rustc_middle/src/mir/consts.rs

```diff
@@ -204,7 +204,7 @@ pub enum Const<'tcx> {
     /// Any way of turning `ty::Const` into `ConstValue` should go through `valtree_to_const_val`;
     /// this ensures that we consistently produce "clean" values without data in the padding or
     /// anything like that.
-    Ty(ty::Const<'tcx>),
+    Ty(Ty<'tcx>, ty::Const<'tcx>),
```

Comment:
I think we should do a bigger change and not add this `Ty<'tcx>`. We should always be able to recompute it should either be used with a `ParamEnv` or is a value.

This is for a separate PR though, so please add a fixme for now.
```suggestion
    // FIXME: Remove this `Ty<'tcx>` again and look up the type for params
    // in the `param_env` instead.
    Ty(Ty<'tcx>, ty::Const<'tcx>),
```

Reply:
Agreed, not really happy with how this basically just skips out on not storing `Ty` for `ty::Const` for this part of the compiler. Will look into `mir::Const::Ty` more after this PR lands